allennlp/semparse/worlds/wikitables_variable_free_world.py

```diff
@@ -0,0 +1,174 @@
+            }
+
+    def __init__(self, table_graph: TableQuestionKnowledgeGraph) -> None:
+        super().__init__(constant_type_prefixes={"string": types.STRING_TYPE,
+                                                 "num": types.NUMBER_TYPE},
+                         global_type_signatures=types.COMMON_TYPE_SIGNATURE,
+                         global_name_mapping=types.COMMON_NAME_MAPPING)
+        self.table_graph = table_graph
+
+        # For every new Sempre column name seen, we update this counter to map it to a new NLTK name.
+        self._column_counter = 0
+
+        # This adds all of the cell and column names to our local name mapping.
+        for entity in table_graph.entities:
+            self._map_name(entity, keep_mapping=True)
+
+        self._entity_set = set(table_graph.entities)
+        self.terminal_productions: Dict[str, str] = {}
+        for entity in self._entity_set:
+            mapped_name = self.local_name_mapping[entity]
+            signature = self.local_type_signatures[mapped_name]
+            self.terminal_productions[entity] = f"{signature} -> {entity}"
+
+        for predicate, mapped_name in self.global_name_mapping.items():
+            if mapped_name in self.global_type_signatures:
+                signature = self.global_type_signatures[mapped_name]
+                self.terminal_productions[predicate] = f"{signature} -> {predicate}"
+
+        # We don't need to recompute this ever; let's just compute it once and cache it.
+        self._valid_actions: Dict[str, List[str]] = None
+
+    def is_table_entity(self, entity_name: str) -> bool:
+        """
+        Returns ``True`` if the given entity is one of the entities in the table.
+        """
+        return entity_name in self._entity_set
+
+    def _get_curried_functions(self) -> Dict[Type, int]:
+        return WikiTablesVariableFreeWorld.curried_functions
+
+    @overrides
+    def get_basic_types(self) -> Set[Type]:
+        return types.BASIC_TYPES
+
+    @overrides
+    def get_valid_actions(self) -> Dict[str, List[str]]:
+        if self._valid_actions:
+            return self._valid_actions
+        valid_actions = super().get_valid_actions()
+        self._valid_actions = valid_actions
+        return valid_actions
+
+    @overrides
+    def get_valid_starting_types(self) -> Set[Type]:
+        return types.STARTING_TYPES
+
+    @overrides
+    def _map_name(self, name: str, keep_mapping: bool = False) -> str:
+        if name not in types.COMMON_NAME_MAPPING and name not in self.local_name_mapping:
+            if not keep_mapping:
+                raise ParsingError(f"Encountered un-mapped name: {name}")
+            if name.startswith("fb:row.row"):
+                # Column name
+                translated_name = "C%d" % self._column_counter
+                self._column_counter += 1
+                self._add_name_mapping(name, translated_name, types.COLUMN_TYPE)
+            elif name.startswith("fb:cell"):
```

Comment:
Do we really want to still use the particular mapping from cells and cell parts to strings that SEMPRE used?  I'm sure that's not what Chen did.  What do we gain from doing this?

Reply:
In fact, looking at the logical forms, I'm pretty sure it'll make the executor more complicated than it needs to be.  We can simplify a bunch of things by removing the SEMPRE naming stuff.

Reply:
I reused the mapping as I wanted to use the `TableQuestionKnowledgeGraph` as is. That's because I think the functionality in that class, such as mapping spans in the question to entities for agenda computation, string normalization etc. are applicable to this world as well. Moreover, cells are typed as strings in the new grammar. It would be nice if the executor can distinguish them from normal strings (when performing string matching in functions such as `filter_in`), and the prefixes from SEMPRE's convention make it easy to do so.
What makes you think that this convention will make the executor more complicated than it needs to be?

Reply:
You'll have to remove prefixes, and do some kind of search to find the appropriate cell or column, because the names are mangled (or store the mapping and give that to the executor, though that still requires changing things).

Reply:
And what's a "normal string" in this language?  When can you produce a string that's not a cell, part, or column?

Also, it'd be nice to shed the crazy `fb:row.row.X` notation.

Reply:
I'd be happy to get rid of SEMPRE notation. But that means changing the `TableKnowledgeGraph`, or adding a similar class for the new language. Seems like an effort that's not worth it.
By normal strings, I essentially meant parts.

Reply:
What strings are allowed in this language?  I just looked at the MAPO paper, and it sure looks like the "strings" always come from the question, not the table.  That seems a lot more sane to me.

Reply:
This seems like a crucial question to me.  I just went through the paper trying to figure this out, and I couldn't find it.  So I sent an email to Chen asking him.  We'll see what he says.

If it's the case that the strings just come from the question, as I expect, then we don't need cells or cell parts at all, and we don't need to select between all of the cells when linking, and we don't have to do the KG embedding.  It simplifies a _whole_ lot of things.

Reply:
Makes sense.